Bio/SeqFeature.py

```diff
@@ -335,6 +335,72 @@ def extract(self, parent_sequence):
                              "sequence file for a valid location.")
         return self.location.extract(parent_sequence)
 
+    def translate(self, parent_sequence, table="Standard", stop_symbol="*",
+            codon_start=1, to_stop=False, cds=False, gap=None):
+        """Get a translation of the feature's sequence.
+
+        This is a shortcut method that will both extract the feature and
+        translate it, taking into account the codon_start and transl_table
+        qualifiers, if they are present. If they are not present the
+        value of the arguments "table" and "codon_start" are used.
+
+        Arguments:
+         - parent_sequence - This method will translate DNA or RNA sequences,
+           and those with a nucleotide or generic alphabet. Trying to
+           translate a protein sequence raises an exception.
+         - table - Which codon table to use if there is no transl_table
+           qualifier for this feature. This can be either a name
+           (string), an NCBI identifier (integer), or a CodonTable
+           object (useful for non-standard genetic codes).  This
+           defaults to the "Standard" table.
+         - codon_start - offset at which the first complete codon of a
```

Comment:
I wonder if we should expose the NCBI one-based ``codon_start`` so directly. Would a Python style zero-based ``start_offset`` make sense (0 or more), defaulting to int(codon_start) - 1.

Reply:
I could remove this option and instead require that codon_start be a qualifier. My thought was that a user may wish to play around with the translation frame but thinking about it, this is really a fix for a corner case in genbank files so perhaps it's better to keep it hidden.

Reply:
I think you were right to expose the starting position, I just think a Python-style offset seems cleaner than the NCBI's 1, 2, 3 values for ``codon_start``.